test/SmartCDNNodeParityTest.php

```diff
@@ -0,0 +1,231 @@
+<?php
+
+namespace transloadit\Test;
+
+use PHPUnit\Framework\TestCase;
+use transloadit\Transloadit;
+
+class SmartCDNNodeParityTest extends TestCase {
+  private Transloadit $transloadit;
+  private string $workspace;
+  private string $template;
+  private string $input;
+  private int $expireAt;
+
+  protected function setUp(): void {
+    if (getenv('TEST_NODE_PARITY') !== '1') {
+      $this->markTestSkipped('Node parity tests are disabled. Set TEST_NODE_PARITY=1 to enable.');
+    }
+
+    if (system('which tsx > /dev/null 2>&1') === false) {
+      $this->markTestSkipped('tsx not available');
+    }
+
+    $key = getenv('TRANSLOADIT_KEY') ?: 'my-key';
+    $secret = getenv('TRANSLOADIT_SECRET') ?: 'my-secret';
+
+    $this->transloadit = new Transloadit([
+      'key' => $key,
+      'secret' => $secret
+    ]);
+
+    $this->workspace = 'my-app';
+    $this->template = 'test-smart-cdn';
+    $this->input = 'inputs/prinsengracht.jpg';
+    $this->expireAt = 1732550672867;
+  }
+
+  private function runNodeScript(array $params): string {
+    $scriptPath = __DIR__ . '/../tool/node-smartcdn-sig.ts';
+
+    $params['auth_key'] = $this->transloadit->key;
+    $params['auth_secret'] = $this->transloadit->secret;
+
+    $jsonInput = json_encode($params);
+
+    $descriptorspec = [
+      0 => ["pipe", "r"],  // stdin
+      1 => ["pipe", "w"],  // stdout
+      2 => ["pipe", "w"]   // stderr
+    ];
+
+    $process = proc_open("tsx $scriptPath", $descriptorspec, $pipes);
+
+    if (!is_resource($process)) {
+      throw new \RuntimeException('Failed to start Node script');
+    }
+
+    fwrite($pipes[0], $jsonInput);
+    fclose($pipes[0]);
+
+    $output = stream_get_contents($pipes[1]);
+    $error = stream_get_contents($pipes[2]);
+
+    fclose($pipes[1]);
+    fclose($pipes[2]);
+
+    $exitCode = proc_close($process);
+
+    if ($exitCode !== 0) {
+      throw new \RuntimeException("Node script failed: $error");
+    }
+
+    return trim($output);
+  }
+
+  public function testBasicUrlGeneration(): void {
+    $params = [
+      'workspace' => $this->workspace,
+      'template' => $this->template,
+      'input' => $this->input,
+      'expire_at_ms' => $this->expireAt
+    ];
+
+    $url = $this->transloadit->signedSmartCDNUrl(
+      $params['workspace'],
+      $params['template'],
+      $params['input'],
+      [],
+      ['expireAtMs' => $params['expire_at_ms']]
+    );
+    $nodeUrl = $this->runNodeScript($params);
+
+    $this->assertEquals($nodeUrl, $url, 'Generated URLs should match node implementation');
```

Comment:
I have the same comment as in the Ruby SDK. Can we also compare the generated URL against a static string here?

Reply:
That's what we have the unit tests for: https://github.com/transloadit/php-sdk/blob/fe2748da1077acb0eb70c803bee8ec444e7dee6c/test/simple/TransloaditTest.php#L149-L178

The parity test is to check it against the reference implementation. When we make changes, it's nice to see if the reference implementation would disagree in CI. After we finalize the reference implementation in the Node SDK, I'll use that SDK to verify parity, instead of this copy pasted function.

Reply:
I now merged parity tests with regular tests to make it a bit more obvious, as well as checking both against hardcoded URLs